lms/djangoapps/courseware/transformers.py

```diff
@@ -0,0 +1,92 @@
+"""
+Grades Transformer
+"""
+from django.contrib.auth.models import AnonymousUser
+from django.test.client import RequestFactory
+
+from openedx.core.lib.block_structure.transformer import BlockStructureTransformer
+from . import module_render
+
+
+class GradesBlockTransformer(BlockStructureTransformer):
+    """
+    The GradesBlockTransformer collects grading information and stores it on
+    the block structure.
+
+    No runtime transformations are performed.
+
+    The following values are stored as xblock_fields on their respective blocks in the
+    block structure:
+
+        due: (datetime) when the problem is due.
+        graded: (boolean)
+        has_score: (boolean)
+        weight: (numeric)
+
+    Additionally, the following value is calculated and stored as a transformer_block_field
+    for each block:
+
+        max_score: (numeric)
+    """
+    VERSION = 1
+    FIELDS_TO_COLLECT = [u'due', u'graded', u'has_score', u'weight']
+
+    @classmethod
+    def name(cls):
+        """
+        Unique identifier for the transformer's class;
+        same identifier used in setup.py.
+        """
+        return u'grades'
+
+    @classmethod
+    def collect(cls, block_structure):
+        """
+        Collects any information that's necessary to execute this
+        transformer's transform method.
+        """
+        block_structure.request_xblock_fields(*cls.FIELDS_TO_COLLECT)
+
+        cls.collect_max_scores(block_structure)
+
+    @staticmethod
+    def _iter_xmodules(block_structure):
+        """
+        Loop through all the blocks locators in the block structure, and retrieve
+        the module (XModule or XBlock) associated with that locator.
+
+        For implementation reasons, we need to pull the max_score from the
+        XModule, even though the data is not user specific.  Here we bind the
+        data to an AnonymousUser.
+        """
+        request = RequestFactory().get('/dummy-collect-max-grades')
+        request.user = AnonymousUser()
+        request.session = {}
+        for block_locator in block_structure.topological_traversal():
+            course_id = unicode(block_locator.course_key)
+            usage_id = block_locator._to_deprecated_string()  # pylint: disable=protected-access
```

Comment:
Change this to `unicode(block_locator)`.  
Otherwise, this doesn't work with Split courses (as found in my integration testing).
By the way, we should never need to call any `to_deprecated_string` methods on Keys.  The Key objects automatically do the right thing.